packages/backend/src/open_payments/auth/middleware.test.ts

```diff
@@ -220,6 +220,42 @@ describe('Auth Middleware', (): void => {
         ]
       })
       const scope = mockAuthServer(grant.toJSON())
+      const next = jest.fn().mockImplementation(async () => {
+        await expect(
+          GrantReference.query().findById(grant.grant)
+        ).resolves.toBeUndefined()
+      })
+      await expect(middleware(ctx, next)).resolves.toBeUndefined()
+      expect(next).toHaveBeenCalled()
+      expect(ctx.grant).toEqual(grant)
+      scope.isDone()
+    }
+  )
+  test.each`
+    type                          | description
+    ${AccessType.IncomingPayment} | ${'incoming payments'}
+    ${AccessType.OutgoingPayment} | ${'outgoing payments'}
+    ${AccessType.Quote}           | ${'quotes'}
+  `(
+    'Stores grant details for $description on the the create path',
+    async ({ type }): Promise<void> => {
```

Comment:
You could test all the actions with something like:
```
  Object.values(AccessAction).forEach((action) => {
    Object.values(AccessType).forEach((type) => {
      if (action === AccessAction.Create) {
        test(`stores grant details for ${type} on the the ${action} path`, async (): Promise<void> => {
        })
      } else {
        test(`does not store grant details for ${type} on the ${action} path`, async (): Promise<void> => {
        })
      }
```

Reply:
Actually, since that :point_up_2: would fail OpenAPI validation, maybe you keep your existing test for all types, but then loop through all actions for just incoming payments.

Reply:
@wilsonianb  we can still test them all like this?
```
  const table: { type: AccessType, action: AccessAction, description: string }[] = []
  Object.values(AccessType).forEach((type) => {
    Object.values(AccessAction).forEach((action) => {
      // The complete path is not valid for outgoing payments
      if(!(type === AccessType.OutgoingPayment && action == AccessAction.Complete) && type !== AccessType.Quote) {
        table.push({
          type: type,
          action: action,
          description: action === AccessAction.Create ? 'stores grant details' : 'does not store grant details'
        })
        // Create and read are the only valid paths for quotes
      } else if(type === AccessType.Quote && (action === AccessAction.Create || action === AccessAction.Read)) {
        table.push({
          type: type,
          action: action,
          description: action === AccessAction.Create ? 'stores grant details' : 'does not store grant details'
        })
      }
    })
  })
```
Or does this make the tests fragile to future changes?